src/rpc/fees.cpp

```diff
@@ -84,7 +85,11 @@ static RPCHelpMan estimatesmartfee()
                 CFeeRate min_mempool_feerate{mempool.GetMinFee()};
                 CFeeRate min_relay_feerate{mempool.m_opts.min_relay_feerate};
                 feeRate = std::max({feeRate, min_mempool_feerate, min_relay_feerate});
-                result.pushKV("feerate", ValueFromAmount(feeRate.GetFeePerK()));
+                if (!request.params[2].isNull() && request.params[2].get_bool()) {
+                    result.pushKV("feerate", static_cast<double>(feeRate.GetFeePerK()) / 1000.0);
```

Comment:
this is wrong, you can't use `double` to denote monetary amounts. `double` is a floating point type, which can not represent decimal values accurately.

For example, if `GetFeePerK()` returns `CAmount{665714}`, the Univalue will be `665.7140000000001`, which is confusing at best, but also wrong.